### PR TITLE
zh: Add doc enable_tls_for_ticdc.md

### LINKED ISSUE #1240: Doc update for https://github.com/pingcap/tidb-operator/pull/3926
## Change Request

**This repository is ONLY used to solve issues related to TiDB Operator documentation.
For other issues (related to TiDB, PD, etc), please move to [other repositories](https://github.com/pingcap/).**

Please answer the following questions before submitting your issue. Thanks!

1. Describe what you find is inappropriate or missing in the existing docs.

2. Describe your suggestion or addition.

3. Provide some reference materials (documents, websites, etc) if you could.

### PR DESCRIPTION
**Pull request by KanShiori**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)
Close #1240

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by KanShiori** [comment]
@DanielZhangQD  PTAL

File: zh/TOC.md [hunk 1]
```diff
@@ -38,6 +38,7 @@
   - [为 MySQL 客户端开启 TLS](enable-tls-for-mysql-client.md)
   - [为 TiDB 组件间开启 TLS](enable-tls-between-components.md)
   - [为 DM 组件开启 TLS](enable-tls-for-dm.md)
+  - [让 TiCDC 组件使用 TLS 访问下游服务](enable_tls_for_ticdc.md)
```
Comment (DanielZhangQD) [inline zh/TOC.md:41]:
```suggestion
  - [TiCDC 组件同步数据到开启 TLS 的下游服务](enable_tls_for_ticdc_sink.md)
```

File: zh/enable_tls_for_ticdc.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+title: 让 TiCDC 组件使用 TLS 访问下游服务
+summary: 了解如何让 TiCDC 使用 TLS 连接访问下游服务
+---
+
+# TiCDC 开启 TLS 访问下游服务
+
+下面主要介绍在 Kubernetes 上如何让 TiCDC 与下游服务同步时使用 TLS 连接。
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc.md:8]:
```suggestion
---
title: TiCDC 组件同步数据到开启 TLS 的下游服务
summary: 了解如何让 TiCDC 组件同步数据到开启 TLS 的下游服务
---

# TiCDC 组件同步数据到开启 TLS 的下游服务

下面主要介绍在 Kubernetes 上如何让 TiCDC 组件同步数据到开启 TLS 的下游服务。
```

File: zh/enable_tls_for_ticdc.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+## 开始之前
+
+你需要部署一个下游服务，并开启了客户端 TLS 认证。
+
+你需要生成客户端访问下游服务所需要的密钥文件。
+
+## TiCDC 使用 TLS 连接访问下游服务
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc.md:16]:
```suggestion
## TiCDC 同步数据到开启 TLS 的下游服务
```

File: zh/enable_tls_for_ticdc.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+1. 创建下游服务客户端 TLS 的 Kubernetes Secret 对象。数据来自于你为客户端生成的密钥文件。
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc.md:18]:
```suggestion
1. 创建包含访问下游服务的客户端 TLS 证书的 Kubernetes Secret 对象。数据来自于你为客户端生成的密钥文件。
```

File: zh/enable_tls_for_ticdc.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create secret generic ${secret_name} --namespace=${cluster_namespace} --from-file=tls.crt=client.pem --from-file=tls.key=client-key.pem --from-file=ca.crt=ca.pem
+    ```
+
+2. 设置 TiDBCluster 定义中的 `spec.ticdc.tlsClientSecretNames` 字段，使得 Secret 对象被挂载到 TiCDC 的 Pod。
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc.md:26]:
```suggestion
2. 设置 TidbCluster 中的 `spec.ticdc.tlsClientSecretNames` 字段，挂载 Secret 对象到 TiCDC 的 Pod。
```

File: zh/enable_tls_for_ticdc.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        # ...
+        tlsClientSecretNames:
+        - ${secret_name}
+   ```
+
+3. 部署集群。
+
+    TiCDC Pod 运行后，你可以在 Pod 内的 `/var/lib/sink-tls/${secret_name}` 目录找到被挂载的密钥文件。
+
+4. 使用下面通过 cdc cli 工具使用 TLS 开启同步任务。
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc.md:48]:
```suggestion
4. 通过 `cdc cli` 工具创建同步任务。
```

File: zh/enable_tls_for_ticdc.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    pd_service=$(kubectl get tc my-tidb-cluster-dev -o=jsonpath='{.status.pd.leader.clientURL}')
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc.md:53]: 这里不用，下面使用 PD service name 就行

File: zh/enable_tls_for_ticdc.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+    kubectl exec ${cluster_name}-ticdc-0 -- /cdc cli changefeed create --pd=http://${pd_service}:2379 - --sink-uri="mysql://${user}:{$password}@${downstream_service}/?ssl-ca=/var/lib/sink-tls/${secret_name}/ca.crt&ssl-cert=/var/lib/sink-tls/${secret_name}/tls.crt&ssl-key=/var/lib/sink-tls/${secret_name}/tls.key"
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc.md:55]:
```suggestion
    kubectl exec ${cluster_name}-ticdc-0 -- /cdc cli changefeed create --pd=http://${pd_service}:2379 --sink-uri="mysql://${user}:{$password}@${downstream_service}/?ssl-ca=/var/lib/sink-tls/${secret_name}/ca.crt&ssl-cert=/var/lib/sink-tls/${secret_name}/tls.crt&ssl-key=/var/lib/sink-tls/${secret_name}/tls.key"
```

File: zh/enable_tls_for_ticdc.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      name: ${cluster_name}
+      namespace: ${cluster_namespace}
+    spec:
+      # ...
+      ticdc:
+        baseImage: pingcap/ticdc
+        version: "v5.0.1"
+        # ...
+        tlsClientSecretNames:
+        - ${secret_name}
```
Comment (csuzhangxc) [inline zh/enable_tls_for_ticdc.md:41]: 如果下游的 TLS 已经是存在我们管理的 secret 里，这里跳过步骤 1，直接在这里引用，可不可以

Comment (KanShiori) [reply zh/enable_tls_for_ticdc.md:41]: 之前的文档应该没有创建对应的 secret？

File: zh/enable_tls_for_ticdc_sink.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+3. 部署集群。
+
+    TiCDC Pod 运行后，你可以在 Pod 内的 `/var/lib/sink-tls/${secret_name}` 目录找到被挂载的密钥文件。
+
+4. 通过 `cdc cli` 工具创建同步任务。
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl exec ${cluster_name}-ticdc-0 -- /cdc cli changefeed create --pd=http://${cluster_name}-pd-0.${cluster_name}-pd-peer:2379 --sink-uri="mysql://${user}:{$password}@${downstream_service}/?ssl-ca=/var/lib/sink-tls/${secret_name}/ca.crt&ssl-cert=/var/lib/sink-tls/${secret_name}/tls.crt&ssl-key=/var/lib/sink-tls/${secret_name}/tls.key"
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc_sink.md:53]:
```suggestion
    kubectl exec ${cluster_name}-ticdc-0 -- /cdc cli changefeed create --pd=https://${cluster_name}-pd:2379 --sink-uri="mysql://${user}:{$password}@${downstream_service}/?ssl-ca=/var/lib/sink-tls/${secret_name}/ca.crt&ssl-cert=/var/lib/sink-tls/${secret_name}/tls.crt&ssl-key=/var/lib/sink-tls/${secret_name}/tls.key"
```

Comment (DanielZhangQD) [reply zh/enable_tls_for_ticdc_sink.md:53]: Use the service name is OK.

File: zh/enable_tls_for_ticdc.md (hunk 9 above)
Comment (csuzhangxc) [reply zh/enable_tls_for_ticdc.md]: 比如 TiDB -> TiCDC -> TiDB，下游的 TiDB 可能就是已经创建好 secret 的了

Comment (DanielZhangQD) [reply zh/enable_tls_for_ticdc.md]: 下游 TiDB 的 secret 正常不会在当前这个集群里面创建吧。

Comment (csuzhangxc) [reply zh/enable_tls_for_ticdc.md]: 嗯，也对。那先不管我这 comment 吧

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

File: zh/enable_tls_for_ticdc.md (hunk 9 above)
Comment (KanShiori) [reply zh/enable_tls_for_ticdc.md]: 因为我们文档都是针对当前集群来说的，没有提及下游集群。所以我觉得还是说明下创建 secret 过程。

File: zh/enable_tls_for_ticdc_sink.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+## 准备条件
+
+在开始之前，请进行以下准备工作：
+
+- 部署一个下游服务，并开启客户端 TLS 认证。
+- 生成客户端访问下游服务所需要的密钥文件。
+
+## TiCDC 同步数据到开启 TLS 的下游服务
+
+1. 创建一个 Kubernetes Secret 对象，此对象需要包含用于访问下游服务的客户端 TLS 证书。数据来自于你为客户端生成的密钥文件。
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc_sink.md:19]:
```suggestion
1. 创建一个 Kubernetes Secret 对象，此对象需要包含用于访问下游服务的客户端 TLS 证书。证书来自于你为客户端生成的密钥文件。
```

**Comment by KanShiori** [comment]
@DanielZhangQD @qiancai  PTAL

File: zh/enable_tls_for_ticdc_sink.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create secret generic ${secret_name} --namespace=${cluster_namespace} --from-file=tls.crt=client.pem --from-file=tls.key=client-key.pem --from-file=ca.crt=ca.pem
+    ```
+
+2. 如果你还未部署 TiDB 集群，在 TidbCluster 定义中添加 `spec.ticdc.tlsClientSecretNames` 字段，然后部署 TiDB 集群。
+
+    如果你已经部署了 TiDB 集群，执行 `kubectl edit tc ${cluster_name}`，并添加 `spec.ticdc.tlsClientSecretNames` 字段，然后等待 TiCDC 的 Pod 自动滚动更新。
```
Comment (DanielZhangQD) [inline zh/enable_tls_for_ticdc_sink.md:29]:
```suggestion
2. 挂载证书文件到 TiCDC Pod

    * 如果你还未部署 TiDB 集群，在 TidbCluster CR 定义中添加 `spec.ticdc.tlsClientSecretNames` 字段，然后部署 TiDB 集群。

    * 如果你已经部署了 TiDB 集群，执行 `kubectl edit tc ${cluster_name} -n ${cluster_namespace}`，并添加 `spec.ticdc.tlsClientSecretNames` 字段，然后等待 TiCDC 的 Pod 自动滚动更新。
```

File: zh/enable_tls_for_ticdc_sink.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+1. 创建一个 Kubernetes Secret 对象，此对象需要包含用于访问下游服务的客户端 TLS 证书。证书来自于你为客户端生成的密钥文件。
+    
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create secret generic ${secret_name} --namespace=${cluster_namespace} --from-file=tls.crt=client.pem --from-file=tls.key=client-key.pem --from-file=ca.crt=ca.pem
+    ```
+
+2. 挂载证书文件到 TiCDC Pod
```
Comment (qiancai) [inline zh/enable_tls_for_ticdc_sink.md:27]:
```suggestion
2. 挂载证书文件到 TiCDC Pod。
```

File: zh/enable_tls_for_ticdc_sink.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+      # ...
+      ticdc:
+        baseImage: pingcap/ticdc
+        version: "v5.0.1"
+        # ...
+        tlsClientSecretNames:
+        - ${secret_name}
+    ```
+
+    TiCDC Pod 运行后，上述创建的 Secret 对象会被挂载到 TiCDC 的 Pod，你可以在 Pod 内的 `/var/lib/sink-tls/${secret_name}` 目录找到被挂载的密钥文件。
```
Comment (qiancai) [inline zh/enable_tls_for_ticdc_sink.md:49]:
```suggestion
    TiCDC Pod 运行后，创建的 Kubernetes Secret 对象会被挂载到 TiCDC 的 Pod。你可以在 Pod 内的 `/var/lib/sink-tls/${secret_name}` 目录找到被挂载的密钥文件。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**